From 6fee9c129e46f689ead84fd0f30dc59cc3b4927b Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Mon, 4 May 2015 10:54:20 +0800
Subject: [PATCH 0141/5242] MLK-9618-5 usb: chipidea: otg: show message device
 no response for SRP fail

commit  9df75d71aaeff3b13d931083209d88fe56bcbd55 from
https://source.codeaurora.org/external/imx/linux-imx.git

If b device does not detect vbus on after TB_SRP_FAIL of data pulse,
b device should show 'Device no response' message.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit 079e1cb81b6259cb4a5e651e1c02f4c3b58a20ed)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/otg_fsm.c   |    1 +
 drivers/usb/common/usb-otg-fsm.c |    1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index 4107c42..3a24585 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -341,6 +341,7 @@ static int b_se0_srp_tmout(struct ci_hdrc *ci)
 static int b_srp_fail_tmout(struct ci_hdrc *ci)
 {
 	ci->fsm.b_srp_done = 1;
+	dev_warn(ci->dev, "Device No Response\n");
 	return 1;
 }
 
diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 0ebaa3d..8eabe5a 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -62,6 +62,7 @@ static void otg_leave_state(struct otg_fsm *fsm, enum usb_otg_state old_state)
 	switch (old_state) {
 	case OTG_STATE_B_IDLE:
 		otg_del_timer(fsm, B_SE0_SRP);
+		otg_del_timer(fsm, B_SRP_FAIL);
 		fsm->b_se0_srp = 0;
 		fsm->adp_sns = 0;
 		fsm->adp_prb = 0;
-- 
1.7.9.5

